pkg/substitute/envsubst.go

```diff
@@ -70,26 +75,36 @@ func SubstituteEnvFromByte(file []byte) ([]byte, error) {
 
 // stripYamlComment takes yaml bytes and returns them back with
 // comments stripped.
-func stripYamlComment(file []byte) []byte {
-	// FIXME use better logic than regex.
-	re := regexp.MustCompile(".?#.*\n")
-	stripped := re.ReplaceAllFunc(file, func(b []byte) []byte {
-		re := regexp.MustCompile("^['\"].+['\"]|^[a-zA-Z0-9]")
-		if re.Match(b) {
-			return b
-		} else {
-			return []byte("\n")
+func stripYamlComment(file []byte) ([]byte, error) {
+	decoder := yaml.NewDecoder(bytes.NewReader(file))
+	var out bytes.Buffer
+	encoder := yaml.NewEncoder(&out)
```

Comment:
Below this line, add
```go
encoder.SetIndent(2)
```
I don't think the end result will be different because at the end of the program run the indentation will be set to `2` anyway but I might be wrong and I want it consistent.